File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    let map = new Map();
    for (let i = 0; i < array.length; i++) {
        if (map.has(array[i])) {
            map.set(array[i], map.get(array[i]) + 1);
        } else {
            map.set(array[i], 1);
        }
    }
    let count = 0;
    let num = 0;
    map.forEach((v, k) => {
        if (v > count) {
            count = v;
            num = k;
        } else if (v === count) {
            num = -1;
        }
    });
    return num;
}

File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.java
import java.util.HashMap;
import java.util.Map;
class Solution {
    public int solution(int[] array) {
        Map< Integer, Integer > map = new HashMap<>();

		for ( int key : array ) {
			map.put( key, map.getOrDefault( key, 0 ) + 1 );
		}

		int count = 0;
		int key = 0;
		for ( Map.Entry< Integer, Integer > item : map.entrySet() ) {
			if ( count < item.getValue() ) {
				count = item.getValue();
				key = item.getKey();
			} else if ( count == item.getValue() ) {
				key = -1;
			}
		}

		return key;
    }
}